Add new tools: Clair, Snyk, Grype

reynico opened the pull request.

reynico committed:
2ms: no dangling files

reynico committed:
tools: install Clair

reynico committed:
tools: install Snyk

reynico committed:
tools: install Grype

reynico committed:
nit: format

reynico committed:
multi: sudo all over the place

reynico commented:
wrong upstream heh

reynico committed:
ci: let's test on develop as well

mattaereal commented on Dockerfile:
```diff
@@ -155,13 +155,29 @@ RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | gpg --
+    && sudo chmod +x /usr/local/bin/snyk
+
+# Install Grype
+RUN ARCH=$(dpkg --print-architecture) \
```
```suggestion
RUN curl -sSfL https://raw.githubusercontent.com/anchore/grype/main/install.sh | sudo sh -s -- -b /usr/local/bin
```

mattaereal commented on Dockerfile:
```diff
@@ -155,13 +155,29 @@ RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | gpg --
+
+# Install Grype
+RUN ARCH=$(dpkg --print-architecture) \
+    && wget -q https://github.com/anchore/grype/releases/download/v0.81.0/grype_0.81.0_linux_$ARCH.deb \
```
```suggestion
```

mattaereal commented on Dockerfile:
```diff
@@ -155,13 +155,29 @@ RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | gpg --
+# Install Grype
+RUN ARCH=$(dpkg --print-architecture) \
+    && wget -q https://github.com/anchore/grype/releases/download/v0.81.0/grype_0.81.0_linux_$ARCH.deb \
+    && sudo dpkg -i grype_0.81.0_linux_$ARCH.deb \
```
```suggestion
```

mattaereal commented on Dockerfile:
```diff
@@ -155,13 +155,29 @@ RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | gpg --
+RUN ARCH=$(dpkg --print-architecture) \
+    && wget -q https://github.com/anchore/grype/releases/download/v0.81.0/grype_0.81.0_linux_$ARCH.deb \
+    && sudo dpkg -i grype_0.81.0_linux_$ARCH.deb \
+    && rm grype_0.81.0_linux_$ARCH.deb
```
```suggestion
```

reynico replied:
y that?

mattaereal commented on Dockerfile:
```diff
@@ -155,13 +155,29 @@ RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | gpg --
+    && sudo chmod +x /usr/local/bin/clair
+
+# Install snyk
+RUN ARCH=$(dpkg --print-architecture) \
```
```suggestion
RUN pnpm install -g snyk
```
I'd suggest installing it directly from the repo, and delete the rest

mattaereal reviewed (changes requested):
I installed grype and snyk differently. Tried to suggest small changes via gh's UI but horrible experience. 

A few PRs after this I successfully installed them both las follows:
`RUN pnpm install -g snyk`
`RUN curl -sSfL https://raw.githubusercontent.com/anchore/grype/main/install.sh | sudo sh -s -- -b /usr/local/bin`

reynico committed:
Update Dockerfile

Co-authored-by: Matías Aereal Aeón <388605+mattaereal@users.noreply.github.com>

reynico committed:
Update Dockerfile

Co-authored-by: Matías Aereal Aeón <388605+mattaereal@users.noreply.github.com>

reynico committed:
Update Dockerfile

Co-authored-by: Matías Aereal Aeón <388605+mattaereal@users.noreply.github.com>

reynico committed:
Update Dockerfile

Co-authored-by: Matías Aereal Aeón <388605+mattaereal@users.noreply.github.com>

reynico committed:
Update Dockerfile

Co-authored-by: Matías Aereal Aeón <388605+mattaereal@users.noreply.github.com>

reynico committed:
dockerfile: remove unneeded script lines

reynico commented:
> I installed grype and snyk differently. Tried to suggest small changes via gh's UI but horrible experience.
> 
> A few PRs after this I successfully installed them both las follows: `RUN pnpm install -g snyk` `RUN curl -sSfL https://raw.githubusercontent.com/anchore/grype/main/install.sh | sudo sh -s -- -b /usr/local/bin`

gotcha! just addressed the comments and committed a last fix to the suggestions. CI is passing as expected so I think this is good to go
